pkg/roachprod/cloud/types/types.go

```diff
@@ -0,0 +1,236 @@
+// Copyright 2025 The Cockroach Authors.
+//
+// Use of this software is governed by the CockroachDB Software License
+// included in the /LICENSE file.
+
+// Package types provides shared types used by both the cloud package and its clients.
+package types
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"regexp"
+	"sort"
+	"text/tabwriter"
+	"time"
+
+	"github.com/cockroachdb/cockroach/pkg/roachprod/config"
+	"github.com/cockroachdb/cockroach/pkg/roachprod/logger"
+	"github.com/cockroachdb/cockroach/pkg/roachprod/promhelperclient"
+	"github.com/cockroachdb/cockroach/pkg/roachprod/ui"
+	"github.com/cockroachdb/cockroach/pkg/roachprod/vm"
+)
+
+const (
+	// The following constants are headers that are used for printing the VM details.
+	headerName           = "Name"
+	headerDNS            = "DNS"
+	headerPrivateIP      = "Private IP"
+	headerPublicIP       = "Public IP"
+	headerMachineType    = "Machine Type"
+	headerCPUArch        = "CPU Arch"
+	headerCPUFamily      = "CPU Family"
+	headerProvisionModel = "Provision Model"
+
+	// Provisional models that are used for printing VM details.
+	spotProvisionModel     = "spot"
+	onDemandProvisionModel = "ondemand"
+)
+
+// printDetailsColumnHeaders are the headers to be printed in the defined sequence.
+var printDetailsColumnHeaders = []string{
+	headerName, headerDNS, headerPrivateIP, headerPublicIP, headerMachineType, headerCPUArch, headerCPUFamily,
+	headerProvisionModel,
+}
+
+// Clusters contains a set of clusters (potentially across multiple providers),
+// keyed by the cluster name.
+type Clusters map[string]*Cluster
+
+// Names returns all cluster names, in alphabetical order.
+func (c Clusters) Names() []string {
+	result := make([]string, 0, len(c))
+	for n := range c {
+		result = append(result, n)
+	}
+	sort.Strings(result)
+	return result
+}
+
+// FilterByName creates a new Clusters map that only contains the clusters with
+// name matching the given regexp.
+func (c Clusters) FilterByName(pattern *regexp.Regexp) Clusters {
+	result := make(Clusters)
+	for name, cluster := range c {
+		if pattern.MatchString(name) {
+			result[name] = cluster
+		}
+	}
+	return result
+}
+
+// A Cluster is created by querying various vm.Provider instances.
+//
+// TODO(benesch): unify with syncedCluster.
```

Comment:
Is this (7+ year old) TODO still relevant? We probably don't have the context for what it fully means anymore, but it seems like after this change we _don't_ want to unify them anymore?

Reply:
Indeed, probably not relevant anymore. Dropping it.